kv/kvserver/reports: TestReplicationStatusReportIntegration failed

kv/kvserver/reports.TestReplicationStatusReportIntegration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9356220?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9356220?buildTab=artifacts#/) on master @ [0b9f1a613e0caad135d9eee7964a0c26f37c563e](https://github.com/cockroachdb/cockroach/commits/0b9f1a613e0caad135d9eee7964a0c26f37c563e):

Fatal error:

```
panic: interface conversion: interface is nil, not error [recovered]
	panic: interface conversion: interface is nil, not error

```
Stack: 

```
goroutine 6083279 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0xc004d9aca8?, {0x5caa328, 0xc010100ed0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:229 +0x6a
panic({0x43b97a0, 0xc0039015c0})
	GOROOT/src/runtime/panic.go:884 +0x212
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).loadOrCreateConnAttempt(0xc015060600, 0x4?, 0xc004d9afc8)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:2718 +0x165
github.com/cockroachdb/cockroach/pkg/rpc.(*Context).VerifyDialback(0xc015060600, {0x5caa328, 0xc010101d70}, 0xc01575f880, 0x40?, {{0xc011907300?, 0x0?, 0xc004d9b0b8?}})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:2676 +0x554
github.com/cockroachdb/cockroach/pkg/server.NewServer.func2({0x5caa328, 0xc010101d70}, 0xc01575f880, 0xc000000003?)
	github.com/cockroachdb/cockroach/pkg/server/server.go:333 +0xf9
github.com/cockroachdb/cockroach/pkg/rpc.(*HeartbeatService).Ping(0xc0110e6300, {0x5caa328, 0xc010101d70}, 0xc01575f880)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/heartbeat.go:173 +0x677
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler.func1({0x5caa328, 0xc010101d70}, {0x483b860?, 0xc01575f880})
	github.com/cockroachdb/cockroach/pkg/rpc/bazel-out/k8-fastbuild/bin/pkg/rpc/rpc_go_proto_/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:354 +0x78
github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor.ServerInterceptor.func1({0x5caa328, 0xc010100ed0}, {0x483b860, 0xc01575f880}, 0xc01657be80, 0xc00c722dc8)
	github.com/cockroachdb/cockroach/pkg/util/tracing/grpcinterceptor/grpc_interceptor.go:115 +0x64a
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5caa328?, 0xc010100ed0?}, {0x483b860?, 0xc01575f880?})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1162 +0x5b
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func3({0x5caa328, 0xc010100ed0}, {0x483b860, 0xc01575f880}, 0xc010100ed0?, 0xc01a2e0000)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:276 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5caa328?, 0xc010100ed0?}, {0x483b860?, 0xc01575f880?})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165 +0x83
github.com/cockroachdb/cockroach/pkg/rpc.kvAuth.unaryInterceptor({0xc001956000?, {{0xc01733d6c0?}, {0x5cca7a0?, 0xc000a07500?}}}, {0x5caa328, 0xc010100ed0}, {0x483b860, 0xc01575f880}, 0xc01657be80, 0xc01a2e0000)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/auth.go:105 +0x2b4
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5caa328?, 0xc010100ed0?}, {0x483b860?, 0xc01575f880?})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165 +0x83
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1.1({0x5caa328?, 0xc010100ed0?})
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:243 +0x39
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunTaskWithErr(0xc004c96780, {0x5caa328, 0xc010100ed0}, {0x7f82e19c13c8?, 0x40?}, 0xc000bbd910)
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:322 +0xd1
github.com/cockroachdb/cockroach/pkg/rpc.NewServerEx.func1({0x5caa328?, 0xc010100ed0?}, {0x483b860?, 0xc01575f880?}, 0x474ae20?, 0x7f82e19c1301?)
	github.com/cockroachdb/cockroach/pkg/rpc/pkg/rpc/context.go:241 +0x95
google.golang.org/grpc.chainUnaryInterceptors.func1.1({0x5caa328?, 0xc010100ed0?}, {0x483b860?, 0xc01575f880?})
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1165 +0x83
google.golang.org/grpc.chainUnaryInterceptors.func1({0x5caa328, 0xc010100ed0}, {0x483b860, 0xc01575f880}, 0xc01657be80, 0xc00c722dc8)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1167 +0x12b
github.com/cockroachdb/cockroach/pkg/rpc._Heartbeat_Ping_Handler({0x42f8300?, 0xc0110e6300}, {0x5caa328, 0xc010100ed0}, 0xc00b475490, 0xc006295300)
	github.com/cockroachdb/cockroach/pkg/rpc/bazel-out/k8-fastbuild/bin/pkg/rpc/rpc_go_proto_/github.com/cockroachdb/cockroach/pkg/rpc/heartbeat.pb.go:356 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc01bef6000, {0x5cf14a0, 0xc0092b4340}, 0xc01639a7e0, 0xc0179b9b90, 0x830a0e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1340 +0xd23
google.golang.org/grpc.(*Server).handleStream(0xc01bef6000, {0x5cf14a0, 0xc0092b4340}, 0xc01639a7e0, 0x0)
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:1713 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:965 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc/external/org_golang_google_grpc/server.go:963 +0x28a
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicationStatusReportIntegration
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/1ce2639b2597bee9434a4c1e166a131c/logTestReplicationStatusReportIntegration1469927895
    test_log_scope.go:79: use -show-logs to present logs inline
```
</p>
</details>
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicationStatusReportIntegration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-26337

--- a/pkg/rpc/context.go
+++ b/pkg/rpc/context.go
@@ -2715,7 +2715,7 @@ func (rpcCtx *Context) loadOrCreateConnAttempt(
 		select {
 		case <-previousAttempt.initialHeartbeatDone:
 			// The connection attempt was completed, return the outcome of it.
-			err := previousAttempt.err.Load().(error)
+			err, _ := previousAttempt.err.Load().(error)
 			if err == nil {
 				// If it completed without error then don't track the connection
 				// anymore. If it did have an error we need to track it until it later gets cleared.

